src/entitysdk/util.py

```diff
@@ -105,7 +110,7 @@ def stream_paginated_request(
             yield data
             number_of_items += 1
 
-            if limit > 0 and number_of_items == limit:
+            if has_limit and number_of_items == limit:
```

Comment:
Wouldn't be enough this?
```suggestion
            if limit and number_of_items == limit:
```

Reply:
What if limit is set to 0.5?

Reply:
The parameter is defined as int or None, so float isn't valid and it would be detected by mypy. I don't think we need to check every possible value or type of parameters that can break the function, but we can guard against reasonable errors. If we want to validate every parameter, than we should use some validator like pydantic (or a typed language :D)

Reply:
While I would agree for most parts of the code, the ones that are directly exposed to the user need a little extra care imo because nothing prevents someone from passing a float. The bad thing here is that the code will work happily in that case instead of breaking.

I am not against using Pydantic validation for the client inputs.

Reply:
I understand your point, but in a library there should be an agreement that the user of the library will respect the the exposed API. If a CLI is built using this sdk, then the input should be properly validated instead.
Do you have examples of other libraries that do type validation in their public methods? Btw I think we can continue the conversation later offline

Reply:
ok, I simplified to what you suggested and we can discuss further validation offline. :)